wa/instruments/perf.py

```diff
@@ -95,6 +95,11 @@ class PerfInstrument(Instrument):
                   description="""Specifies options to be used to gather report when record command
                   is used. It's highly recommended to use perf_type simpleperf when running on
                   android devices as reporting options are unstable with perf"""),
+        Parameter('run_report_sample', kind=bool, default=False, description="""If true, run
+                  'perf/simpleperf report-sample'. It only works with the record command."""),
```

Comment:
What happens if this is used attempted to be used with `stat`? Is it worth adding a check to warn the user if this is that combination is provided?

Reply:
It's ignored if it's used with stat. It's similar to what currently happens if `report_option_string` is used with stat I think. Do you want me to put a warning here (or devlib)?

Reply:
Ah I see, I guess devlib might be the better option then. If you don't mind adding it that would be great thanks.

Reply:
Sure: https://github.com/ARM-software/devlib/pull/545

Reply:
Configuration shouldn't be ignored. This should result in a `ConfigError` (probably something that needs to be added for `report_option_string` as well...)

Reply:
Makes sense. I have added a `validate()` that raises a `ConfigError` if `report_option_string`, `run_report_sample` or `report_sample_options` are specified when the command is not `record`.